Enable to execute Soletta C and FBP test cases in package test (ptest) framework

otcyanglei opened the pull request:
package test (ptest) is convinient way to run unit test in yocto.
This patch is to enable Soletta C/FBP tests in ptest as well.
Referring to https://wiki.yoctoproject.org/wiki/Ptest for ptest format.
- new file "run-ptest":  ptest test execution entry
- modify file "soletta_git.bb": add contents for build and deploy case

Signed-off-by: Lei Yang lei.a.yang@intel.com

barbieri commented on recipes-soletta/soletta/files/add-fbp-passfail-result.patch:
```diff
@@ -0,0 +1,34 @@
+                 logging.warning("Dependence test %s for test %s failed. Skipping it",
+                         curr.depends, curr.test_file)
+                 status["skipped"] += 1
++                logging.info("FAIL: %s" % path["file"])
```
this is not a fail, it's skipped.

otcyanglei replied:
This skip is caused by the dependence test fail. And this is not same as the specified "skipped cases" by user. 
To calculate the pass rate (PASS/all executable cases) in ptest results, I think it should be looked as FAIL though it is skipped actually. 
And I will update this patch and submit tools/fbp-run-tests patch to soletta.

barbieri commented:
the `tools/run-fbp-tests` patch should go to soletta repository, we don't need to keep it in the meta-soletta.

barbieri commented:
the `tools/run-fbp-tests` should go to soletta repository, not meta-soletta.

otcyanglei committed:
Soletta: Enable ptest

package test (ptest) is convinient way to run unit test in yocto.
This patch is to enable Soletta C/FBP tests in ptest as well.
Referring to https://wiki.yoctoproject.org/wiki/Ptest for ptest format.
- new file "run-ptest":  ptest test execution entry
- modify file "soletta_git.bb": add contents for build and deploy case

Signed-off-by: Lei Yang <lei.a.yang@intel.com>

barbieri commented on recipes-soletta/soletta/soletta-ptest.inc:
```diff
@@ -0,0 +1,18 @@
+           "
+
+do_compile_ptest() {
+        oe_runmake CFLAGS="--sysroot=${STAGING_DIR_TARGET} -pthread -lpcre" TARGETCC="${CC}" TARGETAR="${AR}" "tests"
```
why do you need `-pthread -lpcre`?

barbieri replied:
or `--sysroot`

otcyanglei replied:
@barbieri 
By double check, there is no need to add CFLAGS when build test binaries. 
I will amend the patch. Thanks.

mythi commented on recipes-soletta/soletta/soletta_git.bb:
```diff
@@ -87,3 +87,5 @@ do_install() {
    echo "Soletta: $COMMIT_ID" > ${D}/usr/lib/soletta/soletta-image-hash
 }
+
+require soletta-ptest.inc
```
the .inc is not necessary. This can be added to the main .bb.

otcyanglei replied:
amend the patch according to feedback. Thanks.

mythi commented:
Some clean-up's I'd like to see:
- remove all the extra empty lines you are adding
- make commit log subject short (simply e.g., soletta: enable ptest)
- remove all the prefix white spaces in the commit log and ensure the line length is < 75 chars

otcyanglei commented:
@mythi Thanks for review and sharing the commit log guidance. I re-submit the patch according to your feedback. Thanks!

mythi commented:
+1

otcyanglei commented:
Can maintainer help on review and merge the patch? Thanks.

mythi commented:
@otcyanglei please rebase, it gives a merge conflict

otcyanglei commented:
@mythi patch rebase done.

mythi commented:
triggering a CI check

test this please

otcyanglei commented:
@mythi I checked the test log. The fail of the CI test on edison/intel-quark/intel-corei7-64 seems caused by the CI system was trying to get the .dsk.xz image file, which not generated by CI system. Instead, the .dsk or dsk.zip image are in the image download URL. I think the patch did not break CI build. Could you please check?
